### ISSUE TITLE
Meet accessibility requirements in templates

### ISSUE THREAD

**Issue by waylan** [zensical/zensical#781]
### Context

As a test I ran the homepage of an unpublished site through https://www.accessibilitychecker.org/ and found a number of issues, most of which are related to the search dialog. As the search function only uses the defaults, this seems to be an issue with your templates. I am using the 'modern' theme.

### Bug description

* The search `<input>` does not have a label. As explained [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input#:~:text=Placeholders%20are%20not%20accessible), placeholders are not accessible. Either the input needs a label or the [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-label) attribute needs to be defined.
* Two buttons in the search dialog only have a svg as content and so they both need a text label of some sort. Again, an `aria-label` attribute (among other options) would work.
* The `<h3 class="I">Filters</h3>` does not fit into the hierarchical structure of the site. Maybe wrapping it in a `<section>` or `<article>` would resolve that. Not sure.
* Some content is not in defined aria landmark regions of the page (I did not research this one and am not sure what this means or how to resolve it):
    * `<label class="md-overlay" for="__drawer" aria-label="Navigation"></label>`
    * `<div class="e l"><div class="X"><h3 class="I">Filters</h3><h4 class="R">Tags</h4><ol class="F"></ol></div></div>`
* Apparently the scrollable region for search filters is not focusable `<div class="e l"><div class="X"><h3 class="I">Filters</h3><h4 class="R">Tags</h4><ol class="F"></ol></div></div>`

### Related links

- [Report a bug](https://zensical.org/docs/community/contribute/report-a-bug/)
- [Customization](https://zensical.org/docs/customization/)

### Reproduction

No zip file has been provided as there is no way to submit a local page to <https://www.accessibilitychecker.org>.

However, I created a sample repo at [waylan/zensical-sample](https://github.com/waylan/zensical-sample) which is deployed to GitHub pages at <http://waylan.limberg.name/zensical-sample/>. The site only contains the default content created with `zensical new` and no other modifications.

### Steps to reproduce

Pass the URL for any Zensical site with the `modern` theme to <https://www.accessibilitychecker.org>. Warning; without an account you are limited to 2 scans.

A report for the sample site can be found at <https://www.accessibilitychecker.org/audit/?website=http%3A%2F%2Fwaylan.limberg.name%2Fzensical-sample%2F&flag=us>.

A report for my site can be found at <https://www.accessibilitychecker.org/audit/?website=https%3A%2F%2Fwarrenforestpacourt.github.io%2Fwarrenforestcourt.org%2F&flag=us>.

A report for Zensical's own documentation can be found at <https://www.accessibilitychecker.org/audit/?website=https%3A%2F%2Fzensical.org%2Fdocs%2Fget-started%2F&flag=us#>. 

Interestingly, Zensical's documentation has a few different issues being reported due to different config options than me. However, the areas of primary concern are the same. I have not seen a scan of the sample site as I used by 2 free scans on the other 2 sites before creating the sample.

### Before submitting

- [x] I have read and followed the [bug reporting guidelines](https://zensical.org/docs/community/contribute/report-a-bug/).
- [x] I have attached links to [the documentation](https://zensical.org/docs/), and possibly related [issues](https://github.com/zensical/zensical/issues).
- [x] I assure that I have [removed all customizations](https://zensical.org/docs/community/contribute/report-a-bug/#remove-customizations) before submitting this bug report.
- [x] I have ~~attached a __.zip file__~~<insert>linked to a repo</insert>  with a [minimal reproduction](https://zensical.org/docs/community/guides/create-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and welcome to the Zensical community! Really glad to see you here!

Accessibility is definitely an area we need to improve in. Right now, many parts of the UI are still quite raw in that terms, and we plan a bigger overhaul once the foundational functionality of Zensical is stable. We could technically start adding some roles here and there, but solving all issues to make Zensical WCAG compliant isn't something that can be quick-fixed. For this reason, I'll create a backlog item for later consideration once we work on the theme, templates, and search again ☺️

**Comment by squidfunk** [comment]
Thanks for the suggestion! We've added it to our [public backlog] for tracking:

- https://github.com/zensical/backlog/issues/153

We can't commit to a timeline or guarantee implementation, but we do read everything and weigh demand when planning.

### Staying in the loop

- **Subscribe to the backlog item for status changes** – backlog items are locked to keep notifications focused.
- **Watch the [public backlog repository]** – subscribe to the repo to get notified when we add or update items.
- **Start a discussion in [Zensical Spark]** – if you use Zensical professionally, Zensical Spark is the best place to flesh out the details together with us and signal demand.

### How we work

Curious about how we prioritize and implement changes? Our [how we work] guide covers the full process, including how we select requests and work with the community.

[public backlog]: https://github.com/orgs/zensical/projects/2/views/1
[public backlog repository]: https://github.com/zensical/backlog
[Zensical Spark]: https://zensical.org/spark/
[how we work]: https://zensical.org/docs/community/how-we-work/
